recipes/gdptools/tests/test_RWE_Class.py

```diff
@@ -0,0 +1,138 @@
+"""Testing ancillary functions."""
+from typing import Any
+
+import geopandas as gpd
+import numpy as np
+import numpy.typing as npt
+import pandas as pd
+import pytest
+import xarray as xr
+from pyproj.crs import CRS
+from pyproj.exceptions import CRSError
+
+from gdptools.ancillary import _check_for_intersection
+from gdptools.ancillary import _generate_weights_pershp
+from gdptools.ancillary import _get_cells_poly
+from gdptools.ancillary import _get_crs
+from gdptools.ancillary import _get_data_via_catalog
+from gdptools.ancillary import _get_shp_file
+from gdptools.gdp_data_class import CatGrids
+from gdptools.gdp_data_class import CatParams
+
+
+@pytest.mark.parametrize(
```

Comment:
in the longer-term, these should probably move to the project instead of this recipe, right?

Reply:
Yep. I'm planning on sending a PR upstream so we can remove this from here.

Reply:
That is why I added the tests. I relaxed all those that I could and ensure a passing build. The others will need some upstream modifications.

Yeah, folks usually confuse apps with libraries.